src/particles/spacecharge/GatherAndPush.cpp

```diff
@@ -35,6 +38,23 @@ namespace impactx::particles::spacecharge
 
         amrex::ParticleReal const charge = pc.GetRefParticle().charge;
 
+        // Deposit 1D charge density in cases where it is required.
+        amrex::Real * beam_profile = nullptr;
+        amrex::Real * beam_profile_slope = nullptr;
```

Comment:
Allocate the profiles here. Note that `_slope` is `num_bins` or so and the charge is `num_bins + 1`.
```suggestion
        amrex::Gpu::DeviceVector<amrex::Real> beam_profile(num_bins + 1, 0.0);
        amrex::Gpu::DeviceVector<amrex::Real> beam_profile_slope(num_bins, 0.0);
```